stock_orderpoint_generator/models/orderpoint_template.py

```diff
@@ -42,32 +86,88 @@ class OrderpointTemplate(models.Model):
     )
     auto_last_generation = fields.Datetime(string='Last Automatic Generation')
 
+    def _template_fields_to_discard(self):
+        """In order to create every orderpoint we should pop this template
+           customization fields """
+        return [
+            'auto_generate', 'auto_product_ids', 'auto_last_generation',
+            'auto_min_qty', 'auto_min_date_start', 'auto_min_qty_criteria',
+            'auto_min_date_end', 'auto_max_date_start', 'auto_max_date_end',
+            'auto_max_qty_criteria', 'auto_max_qty',
+        ]
+
     def _disable_old_instances(self, product_ids):
-        """ Clean old instance by setting those inactives
-        """
+        """Clean old instance by setting those inactives"""
         orderpoints = self.env['stock.warehouse.orderpoint'].search(
-            [('product_id', 'in', product_ids)]
+            [('product_id', 'in', product_ids.ids)]
         )
         orderpoints.write({'active': False})
 
+    @api.model
+    def _get_product_qty_by_criteria(
+            self, product_ids, location_id, from_date, to_date, criteria):
+        """Returns a dict with product ids as keys and the resulting
+           calculation of historic moves according to criteria"""
+        stock_qty_history = product_ids._compute_historic_quantities_dict(
+            location_id=location_id,
+            from_date=from_date,
+            to_date=to_date)
+        criteria_list = [x[0] for x in self._get_auto_criteria_selection()]
+        if criteria not in criteria_list:
+            return {x: 0 for x, y in stock_qty_history.items()}
+        if criteria == 'max':
```

Comment:
Instead of this ugly code, use a dictionary to map function callers and call it after:
```python
criteria_methods = {
    'max': max,
    'median': median_high,
    'avg': mean,
    'min': min,
}
return {x: criteria_methods[criteria](y['stock_history']) for x,y in stock_qty_history.items()}
```